Tests/UriUnitTests/UriUnitTests.cs

```diff
@@ -0,0 +1,716 @@
+using nanoFramework.TestFramework;
+using System;
+
+namespace System.Http
+{
+    [TestClass]
+    public class UriUnitTests
+    {
+        Uri uri;
+        UriProperties props;
+        ParsedUri parsed;
+
+        [TestMethod]
+        public void TestCtor_String()
+        {
+            Uri uri = new Uri(@"http://foo/bar/baz#frag");
+
+            Assert.Equal(@"http://foo/bar/baz#frag", uri.AbsoluteUri, $"expecting {@"http://foo/bar/baz#frag"}, got {uri.AbsoluteUri}");
+
+            // TODO needs fix
```

Comment:
what needs to be fixed and where? That can help someone who wants to pick it up.
I see other TODO, would be great to add a bit of context